File: marketing master/android/mm2/app/src/main/java/com/scrat/mm2/WakeUpActivity.java
package com.scrat.mm2;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.xinstall.XInstall;
import com.xinstall.listener.XWakeUpAdapter;
import com.xinstall.model.XAppData;

import java.util.Map;

public class WakeUpActivity extends AppCompatActivity {

    private View backBtn;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_wakeup);
        backBtn = findViewById(R.id.back);
        /**
         *  获取唤醒参数
         */
        XInstall.getWakeUpParam(this,getIntent(), wakeUpAdapter);
        initView();
    }

    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);
        /**
         *  获取唤醒参数
         */
        XInstall.getWakeUpParam(this,getIntent(), wakeUpAdapter);
    }

    private void initView() {
        backBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onBackPressed();
            }
        });
    }

    public void onBackPressed() {
        super.onBackPressed();
        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
    }

    XWakeUpAdapter wakeUpAdapter = new XWakeUpAdapter() {
        @Override
        public void onWakeUp(XAppData xAppData) {
            // 获取渠道数据
            String channelCode = xAppData.getChannelCode();
            Map<String, String> extraData = xAppData.getExtraData();
            String timeSpan = xAppData.getTimeSpan();

            String str = "channelCode = "+ channelCode +
                         "\nco = " + extraData.get("co") +
                         "\nuo = " + extraData.get("uo") +
                         "\ntimeSpan = " + timeSpan;
            TipDialogSingleButton dialog = new
                    TipDialogSingleButton(WakeUpActivity.this);
            dialog.setTitle("唤醒参数示例").setMessage(str).setYesOnclickListener(new TipDialogSingleButton.onYesOnclickListener() {
                @Override
                public void onYesClick() {
                    dialog.dismiss();
                }
            }).show();
        }

    };
}

File: XinstallAndroidDemoApp_1.1.3/xinstalldemo/src/main/java/com/scrat/mm2/sample/WakeUpActivity.kt
package com.scrat.mm2.sample

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.scrat.mm2.R
import com.xinstall.XInstall
import com.xinstall.listener.XWakeUpAdapter
import com.xinstall.model.XAppData
import kotlinx.android.synthetic.main.activity_channel.*

class WakeUpActivity : AppCompatActivity() {

    companion object {
        fun start(context: Context) {
            val intent = Intent(context, WakeUpActivity::class.java)
            context.startActivity(intent)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_wakeup)

        /**
         *  获取唤醒参数 
         */
        
        XInstall.getWakeUpParam(getIntent(), wakeUpAdapter)

        initView()
    }

    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)

        /**
         *  获取唤醒参数
         */
        XInstall.getWakeUpParam(getIntent(), wakeUpAdapter)
    }

    private fun initView() {
        back.setOnClickListener {
            onBackPressed()
        }
    }

    override fun onBackPressed() {
        super.onBackPressed()
        val intent = Intent(this, SimpleMainActivity::class.java)
        startActivity(intent)
    }

    private var wakeUpAdapter: XWakeUpAdapter? = object : XWakeUpAdapter() {
        override fun onWakeUp(xAppData: XAppData) {
            // 获取渠道数据
            val channelCode = xAppData.getChannelCode()
            val extraData = xAppData.extraData
            val timeSpan = xAppData.timeSpan

            var str =
                "channelCode=${xAppData!!.channelCode}\nco=${extraData["co"]}\nuo=${extraData["uo"]}\ntimeSpan=${xAppData!!.timeSpan}"
            val dialog =
                TipDialogSingleButton(this@WakeUpActivity)
            dialog.setTitle("唤醒参数示例").setMessage(str)
                .setYesOnclickListener {
                    dialog.dismiss()
                }.show()
        }

    }
}
